core/include/userver/engine/mutex.hpp

```diff
@@ -59,11 +59,9 @@ class Mutex final {
   bool try_lock_until(Deadline deadline);
 
  private:
-  bool LockFastPath(impl::TaskContext&);
-  bool LockSlowPath(impl::TaskContext&, Deadline);
+  class ImplWrapper;
 
-  std::atomic<impl::TaskContext*> owner_;
-  impl::FastPimplWaitList lock_waiters_;
+  utils::FastPimpl<ImplWrapper, impl::kWaitListSize + 8, alignof(void*)> impl_wrapper_;
```

Comment:
The mix of an integer and `alignof(void*)` on the same line does not look good. `sizeof(impl::TaskContext*)` might be slightly better.